src/main/kotlin/calculator/Calculator.kt

```diff
@@ -0,0 +1,59 @@
+package calculator
+
+import java.math.BigDecimal
+import java.util.LinkedList
+import java.util.Queue
+
+class Calculator {
+
+    companion object {
+        const val OPERAND = "-?\\d+"
+        const val OPERATOR = "[+\\-*/]"
+        const val DELIMITER = " "
+    }
+
+    fun exec(str: String): BigDecimal {
+        if (str.matches(Regex("$OPERAND(${DELIMITER}${OPERATOR}${DELIMITER}$OPERAND)+"))) {
+            return calculate(operators(str), operands(str))
+        }
+        throw IllegalArgumentException("${str}은 올바른 수식이 아닙니다.")
```

Comment:
early return  패턴 👍 
하지만 exec 함수의 메인 동작은 예외처리인 throw가 아닌 calculate가 아닐까요?
```suggestion
        if (!str.matches(Regex("$OPERAND(${DELIMITER}${OPERATOR}${DELIMITER}$OPERAND)+"))) {
            throw IllegalArgumentException("${str}은 올바른 수식이 아닙니다.")
        }
        return calculate(operators(str), operands(str))  
```
가 조금더 자연스러워 보이진 않을까요?